tests/mixin.flowchart.tests.js

```diff
@@ -0,0 +1,87 @@
+
+class FlowchartItem extends mix(InMemoryEntity).with(FlowchartItemMixin) {}
+
+describe("Flowchart Mixins", () => {
+    it("flowchart item can be added", () => {
+        const entity = new FlowchartEntity();
+        entity.addUnit(new FlowchartItem());
+        expect(entity.units.length).to.be.equal(1);
+    });
+
+    it("flowchart item can be added inbetween", () => {
+        const entity = new FlowchartEntity();
+        entity.addUnit(new FlowchartItem());
+        entity.addUnit(new FlowchartItem());
+        expect(entity.units[0].next).to.be.equal(entity.units[1].flowchartId);
+        const item = new FlowchartItem();
+        entity.addUnit(item, 1);
+        expect(entity.units[1].flowchartId).to.be.equal(item.flowchartId);
+        expect(entity.units[1].next).to.be.equal(entity.units[2].flowchartId);
+    });
+
+    it("flowchart item points to correct target after removalof inbetween unit", () => {
+        const entity = new FlowchartEntity();
+        entity.addUnit(new FlowchartItem());
+        entity.addUnit(new FlowchartItem());
+        entity.addUnit(new FlowchartItem());
+        entity.removeUnit(entity.units[1].flowchartId);
+        expect(entity.units[0].next).to.be.equal(entity.units[1].flowchartId);
+    });
+
+    it("flowchart item points to correct target after removal of last unit", () => {
+        const entity = new FlowchartEntity();
+        entity.addUnit(new FlowchartItem());
+        entity.addUnit(new FlowchartItem());
+        entity.addUnit(new FlowchartItem());
+        entity.removeUnit(entity.units[2].flowchartId);
+        // eslint-disable-next-line no-unused-expressions
+        expect(entity.units[1].next).to.be.null;
+    });
+
+    it("flowchart item can be removed", () => {
+        const entity = new FlowchartEntity();
+        const item = new FlowchartItem();
+        entity.addUnit(item);
+        entity.removeUnit(item.flowchartId);
+        expect(entity.units.length).to.be.equal(0);
+    });
+
+    it("flowchart item can be replaced inbetween", () => {
+        const entity = new FlowchartEntity();
+        entity.addUnit(new FlowchartItem());
+        entity.addUnit(new FlowchartItem());
+        entity.addUnit(new FlowchartItem());
+        const unitsBeforeReplace = entity.units.length;
+        const item = new FlowchartItem();
+        entity.replaceUnit(item, 1);
+        expect(entity.units[1].flowchartId).to.be.equal(item.flowchartId);
+        expect(entity.units.length).to.be.equal(unitsBeforeReplace);
+    });
+
+    it("flowchart item can be accessed by flowchart Id", () => {
+        const entity = new FlowchartEntity();
+        entity.addUnit(new FlowchartItem());
+        entity.addUnit(new FlowchartItem());
+        const item = new FlowchartItem();
+        entity.addUnit(item);
+        const fetchedUnit = entity.getUnit(item.flowchartId);
+        expect(fetchedUnit._json).to.be.deep.equal(item._json);
```

Comment:
A couple of thoughts:

1. Should we really be doing a deep compare here? Isn't the whole purpose to get the actual unit - i.e. class instance?
2. For the deep comparison of entities, there might be a util function (if not, probably should be)

Reply:
1. Ok, I will add `expect(fetchedUnit).to.be.instanceOf(FlowchartItem)`. I think besides the type we should also check for the content, i.e. at least the `flowchartId`, in order to make sure that the correct unit was retrieved.
2. If the entities are JS types, we can use the functions implemented by chai, e.g. `equal`, `eql`, `deep`, so I don't think a utility function is necessary.